habitat-lab/habitat/tasks/rearrange/actions/pddl_actions.py

```diff
@@ -71,26 +72,31 @@ def step(self, *args, is_last_action, **kwargs):
                         raise ValueError(
                             f"Got invalid action value < 0 in {action_part} with action {action}"
                         )
-                rearrange_logger.debug(f"Got action part {real_action_idxs}")
 
                 param_values = [
                     self._entities_list[i] for i in real_action_idxs
                 ]
 
-                apply_action = action.copy()
+                apply_action = action.clone()
                 apply_action.set_param_values(param_values)
+                self._prev_action = apply_action
                 if self._task.pddl_problem.is_expr_true(apply_action.precond):
-                    rearrange_logger.debug(
-                        f"Applying action {action} with obj args {param_values}"
-                    )
                     self._task.pddl_problem.apply_action(apply_action)
                 else:
-                    rearrange_logger.debug(
-                        f"Preconds not satisfied for: action {action} with obj args {param_values}"
-                    )
                     self._was_prev_action_invalid = True
 
             cur_i += action.n_args
+
+    def step(self, *args, is_last_action, **kwargs):
```

Comment:
Hm I don't like anything other than an 'env' class having a 'step' function. Why can't we have this class result the action to be applied, and then have an env apply the action?

Reply:
I agree, but this is how all actions in Habitat work (beyond this PR). Let's address in future PR.